### PR TITLE
llvm: Update to 11.0.0

### PR DESCRIPTION
**Pull request by Snektron**
Update llvm and its subprojects to 11.0.0.

There were a few details i solved as follows:
* For llvm11, files/llvm-Config-config.h was not used, so i removed it. Same for llvm10
* I just copied all patches from llvm10, which yielded a few problems:
  - Some code near clang-004-ppc64-musl-elfv2.patch got changed so i re-made the patch
  - llvm-007-ppc-d85007.patch [seems to have been implemented upstream](https://github.com/llvm/llvm-project/commit/cbea17568f4301582c1d5d43990f089ca6cff522)
* subpackages (clang, lld, lldb, etc) were removed from llvm10 as they are provided by llvm11 now 
* llvm-unwind(-devel) and libcxx(-devel) were updated to 11.0.0
* There was a [sed expression](https://github.com/void-linux/void-packages/blob/4712555551b5140760a17948a5c38eac1cf13ddc/srcpkgs/libcxx/template#L27) in the extract step for libcxx, it seems that this has been changed upstream so i removed it.

### CODE REVIEW

File: srcpkgs/libcxx/template [hunk 1]
```diff
@@ -1,31 +1,41 @@
...
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="NCSA, MIT"
 homepage="https://libcxx.llvm.org"
-distfiles="https://github.com/llvm/llvm-project/releases/download/llvmorg-${version}/libcxx-${version}.src.tar.xz"
-checksum=270f8a3f176f1981b0f6ab8aa556720988872ec2b48ed3b605d0ced8d09156c7
+distfiles="
+ https://github.com/llvm/llvm-project/releases/download/llvmorg-${version}/libcxx-${version}.src.tar.xz
+ https://github.com/llvm/llvm-project/releases/download/llvmorg-${version}/libcxxabi-${version}.src.tar.xz
+ https://github.com/llvm/llvm-project/releases/download/llvmorg-${version}/llvm-${version}.src.tar.xz
+"
```
Comment (fosslinux) [inline srcpkgs/libcxx/template:21]: Put quote on previous line

File: srcpkgs/libcxx/template [hunk 2]
```diff
@@ -1,31 +1,41 @@
...
+ 913f68c898dfb4a03b397c5e11c6a2f39d0f22ed7665c9cefa87a34423a72469"
 
 case "$XBPS_TARGET_MACHINE" in
 	*-musl) configure_args+=" -DLIBCXX_HAS_MUSL_LIBC=YES"
 esac
 
 post_extract() {
 	case "$XBPS_TARGET_MACHINE" in
-	*-musl) patch -Np1 <${FILESDIR}/libcxx-0001-musl-hacks.patch
-		sed -i 's/!defined(_GCC_MAX_ALIGN_T)/& \&\& !defined(__DEFINED_max_align_t)/' include/stddef.h ;;
```
Comment (fosslinux) [inline srcpkgs/libcxx/template:27]: Remind myself to check whether this is correct

File: srcpkgs/libcxxabi/template [hunk 3]
```diff
@@ -27,10 +27,8 @@ esac
 
 post_extract() {
 	# Break dependency cycle, we just need the headers here
```
Comment (fosslinux) [inline srcpkgs/libcxxabi/template:29]: Get rid of the comment too

File: srcpkgs/llvm11/files/patches/llvm/llvm-002-musl-ppc64-elfv2.patch [hunk 4]
```diff
@@ -0,0 +1,26 @@
+This patches LLVM to use ELFv2 on ppc64 uncoditionally unless overridden. We
```
Comment (fosslinux) [inline srcpkgs/llvm11/files/patches/llvm/llvm-002-musl-ppc64-elfv2.patch:1]: unconditionally

File: srcpkgs/llvm11/files/patches/llvm/llvm-003-secureplt.patch [hunk 5]
```diff
@@ -0,0 +1,11 @@
+--- llvm/lib/Target/PowerPC/PPCSubtarget.cpp
```
Comment (fosslinux) [inline srcpkgs/llvm11/files/patches/llvm/llvm-003-secureplt.patch:1]: Maybe `llvm-003-ppc-secureplt.patch` would be a better name

**Review by fosslinux — CHANGES_REQUESTED** [review]
An impressive first PR!

**Comment by fosslinux** [comment]
Also lowercase `update` rather than `Update` is the convention but dosen't really matter. The llvm10 commit should be `llvm10: rebuild standalone.`.

File: srcpkgs/libcxx/template (hunk 2 above)
Comment (Snektron) [reply srcpkgs/libcxx/template:27]: The line which this sed expression is supposed to change does no longer exist in the relevant file. What should i put here?

Comment (fosslinux) [reply srcpkgs/libcxx/template:27]: Nothing probably, I'm just looking for anything similar

**Comment by fosslinux** [comment]
Rather than `bump revision` a better commit message would be `rebuild for llvm11`. Also, before this is merged, all the commits will need to be squashed.

**Comment by q66** [comment]
what? no, don't squash rebuild commits please

**Comment by Snektron** [comment]
All that seems to be left are Crystal and Rust. Crystal does not seem to be llvm 11 compatible yet (see https://github.com/crystal-lang/crystal/pull/9829), so i think this should be about ready.

**Comment by Snektron** [comment]
What should i do about this CI?

**Comment by fosslinux** [comment]
Add `[ci skip]` to the last commit.

**Pull request closed without merging**